src/Robot.elm

```diff
@@ -0,0 +1,89 @@
+module Main exposing (Model, Msg(..), init, main, subscriptions, update, view)
+
+import Browser
+import Browser.Events
+import Html exposing (Html, button, div, text)
+import Html.Attributes as Attributes exposing (style)
+import Html.Events exposing (onClick)
+import Json.Decode
+
+main =
+    Browser.element { init = init, update = update, view = view, subscriptions = subscriptions }
+
+
+type alias Model =
+    { xblue : Int, yblue : Int , xred : Int, yred : Int}
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( { xblue = 50, yblue = 50 ,xred = 10, yred = 10}, Cmd.none )
+
+
+type Msg
+    = KeyDown Int
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Browser.Events.onKeyDown (Json.Decode.map KeyDown Html.Events.keyCode)
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        KeyDown code ->
+            case code of
+                37 ->
+                    ( { model | xblue = model.xblue - 1 }, Cmd.none )
+
+                38 ->
+                    ( { model | yblue = model.yblue - 1 }, Cmd.none )
+
+                39 ->
+                    ( { model | xblue = model.xblue + 1 }, Cmd.none )
+
+                40 ->
+                    ( { model | yblue = model.yblue + 1 }, Cmd.none )
+
+                65 ->
+                    ( { model | xred = model.xred - 1 }, Cmd.none )
+
+                87 ->
+                    ( { model | yred = model.yred - 1 }, Cmd.none )
+
+                68 ->
+                    ( { model | xred = model.xred + 1 }, Cmd.none )
+
+                83 ->
+                    ( { model | yred = model.yred + 1 }, Cmd.none )
+
+                default ->
+                    ( model, Cmd.none )
+
+
+view : Model -> Html.Html Msg
+view model =
+    Html.div []
+        [ Html.div
+            [ style "background-color" "blue"
+            , style "color" "yellow"
+            , style "width" "2%"
+            , style "height" "3%"
+            , style "text-align" "center"
+            , style "position" "absolute"
+            , style "left" (String.fromInt model.xblue ++ "%")
+            , style "top" (String.fromInt model.yblue ++ "%")

```

Comment:
When you have trailing parentheses, like here, you should prefer a pipe:
```elm
, style "left" <| String.fromInt model.xblue ++ "%"
```
is just like
```elm
, style "left" (String.fromInt model.xblue ++ "%")
```
but is nicer when you get used to it.